### DISCUSSION TITLE
how to place macro

### DISCUSSION THREAD

**Discussion by SJA0000** [The-OpenROAD-Project/OpenROAD-flow-scripts#2318] [Q&A]
Hello
I'm working with Openroad-flow-scripts, and i wanted to use place_macro.
I've implemented our design and would like to place multiple of this module and obtain power and area results. 
when i run the ORFS, the flow stops immediately. 
I'm new with this work, so could you help me how to use macro_placement properly?
 

Below is the error message: 

Commented 0 lines containing "original_pin"
Replaced malformed functions 0
Writing replaced file: objects/asap7/my_design/base/lib/asap7sc7p5t_SEQ_RVT_FF_nldm_220123.lib
Makefile:183: /home/jina/OpenROAD-flow-scripts/flow/platforms//config.mk: No such file or directory
make[2]: *** No rule to make target '/home/jina/OpenROAD-flow-scripts/flow/platforms//config.mk'.  Stop.
make[1]: *** [Makefile:366: results/asap7/my_design_entry1/base/entry1.lib] Error 2
make: *** [Makefile:497: results/asap7/my_design/base/1_1_yosys.v] Error 2

**Comment by mi-and-core** [comment]
Two small remarks:
(1) "Makefile:183: ..."  indicates that the Makefile used is older. I would have expected a "Makefile:204: ..."
(2) Have you set "export PLATFORM=asap7" in the config.mk of your design?

**Comment by maliberty** [comment]
platforms//config.mk shows that the platform is unset as its name should be between the //

**Reply by SJA0000** [reply]
Thanks!
but i already set the platform. 
I change command "export BLOCKS = " to "export BLOCK = ", then it works. Did i do correctly..? 

However, another problem occured.. 

I'm planing to place my macro at the center of the Die with other mock blocks, and before that, i'm trying to place one macro for practice. 
The followings are the config file and tcl file... OpenROAD-flow-scripts works but i obtained unexpacted result... 

![image](https://github.com/user-attachments/assets/539a3bda-72a2-470d-b8da-c4c2c740ac2e)
![image](https://github.com/user-attachments/assets/9df5cb57-3eac-4b1a-8e8d-b64ecae5e72b)


Could you please advise how to fix this outcome to look like a compact block with clear boundary? 

config.mk
```
export PLATFORM               = asap7
export DESIGN_NAME            = bts_pinplace

export VERILOG_FILES          = ./designs/src/xbar_jina/$(DESIGN_NAME).v \
								./designs/src/xbar_jina/entry1.v
export SDC_FILE               = ./designs/$(PLATFORM)/1bts_test/constraint.sdc


export DIE_AREA = 0 0 4950 220
export CORE_AREA = 5 5 1925 205
export PLACE_DENSITY_LB_ADDON = 0.10
export CORE_MARGIN       = 2

#to place a macro at the center of the Die 
export BLOCK = entry1
export MACRO_PLACEMENT_TCL   = ./designs/$(PLATFORM)/1bts_test/macro-placement.tcl

export SYNTH_HIERARCHICAL = 1
export RTLMP_FLOW = True 

export MACRO_HALO_X = 5
export MACRO_HALO_Y = 5

export MAX_UNGROUP_SIZE ?= 0.1
export PLACE_DENSITY            = 0.45
``` 

macro-placement.tcl

`place_macro -macro_name unit0 -location {2470 110}`

I'd greatly appreciate any help you can offer.

**Comment by maliberty** [comment]
The macro appears to be centered in the die box which extends far to the right.
